tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -13,566 +13,566 @@
 class NodeScopeResolverTest extends TypeInferenceTestCase
 {
 
-	public function dataFileAsserts(): iterable
+	public function dataAssertions(): iterable
```

Comment:
I am wondering whether we can adjust this PR in a way that already exisiting `TypeInferenceTestCase` subclasses immediately benefit from the base-class changes without modifications..?

(Do we need this new mechanism in parallel to the pre-exisiting one?)

Reply:
didn't figure that out yet, I think it's not possible, but maybe I'm missing some magic to make it possible :) the problems are different return and argument types for the 2 methods

Reply:
wondering still about the same thing. couldn't you rename the `gatherAssertTypes` method which existed before this PR to e.g. `gatherAssertTypesFromFile` and implement the PRs `gatherAssertions` as `gatherAssertTypes` (same for `assertFileAsserts`)  and then we can take this changes as a drop-in replacement without needs for change in `NodeScopeResolver` test..

maybe I am missing something obvious?

Reply:
I just realized that the thing which is not compatibel is https://github.com/phpstan/phpstan-src/pull/1991/files#diff-1e9b47d3e7051caa11f9d76ff7bd6ff2afb430014dd93dad916cbb8c2148f545R1139